lib/PuppeteerSharp.Tests/PageTests/SetContentTests.cs

```diff
@@ -128,5 +129,49 @@ public async Task ShouldWorkWithNewline()
             await Page.SetContentAsync("<div>\n</div>");
             Assert.Equal("\n", await Page.QuerySelectorAsync("div").EvaluateFunctionAsync<string>("div => div.textContent"));
         }
+
+        [Fact]
+        public async Task ShouldWorkWithoutThrowingUnhandledTaskExceptions()
+        {
+            // Using this technique to test finalizers:
+            // https://www.inversionofcontrol.co.uk/unit-testing-finalizers-in-csharp/
+
+            TaskScheduler.UnobservedTaskException += TaskScheduler_UnobservedTaskException;
+
+            WeakReference<Page> weak = null;
+
+            Func<Task> func = async () =>
+            {
+                var page = await Browser.NewPageAsync();
+                await page.SetContentAsync("<html></html>");
+                await page.CloseAsync();
+
+                weak = new WeakReference<Page>(page, true);
+            };
+
+            await func();
+            GC.Collect(0, GCCollectionMode.Forced);
+            GC.WaitForPendingFinalizers();
+
+            Assert.Empty(exceptions);
+
+            TaskScheduler.UnobservedTaskException -= TaskScheduler_UnobservedTaskException;
+        }
+
+        private static List<TargetClosedException> exceptions = new List<TargetClosedException>();
+
+        private void TaskScheduler_UnobservedTaskException(object sender, UnobservedTaskExceptionEventArgs args)

```

Comment:
We can move the event inside the test. Like we do here https://github.com/kblok/puppeteer-sharp/blob/master/lib/PuppeteerSharp.Tests/PageTests/Events/ConsoleTests.cs#L21